subsys/bluetooth/host/smp.c

```diff
@@ -2775,6 +2775,11 @@ static u8_t smp_pairing_req(struct bt_smp *smp, struct net_buf *buf)
 	    (req->auth_req & BT_SMP_AUTH_BONDING)) {
 		atomic_set_bit(smp->flags, SMP_FLAG_BOND);
 	}
+#if defined(CONFIG_BT_BONDING_REQUIRED)
```

Comment:
I guess this can be replaced by else if (IS_ENABLED(CONFIG_BT_BONDING_REQUIRED))

Reply:
Yes indeed, if you prefer such semantics.

Reply:
> if you prefer such semantics.

@demantmrrv it sounds like you may not be aware of the benefits of `IS_ENABLED`?

 - #ifdefs generally make the code harder to read (you're forced to interpret two languages that operate on different levels: C and preprocessor). Using `IS_ENALBED()` transforms the conditionals to standard C syntax.
 - Using #ifdefs hides errors that the compiler would catch if it was exposed to the code in question. Using `IS_ENABLED()` always exposes the code to the compiler so it can catch such issues even if the code doesn't end up in the final image.

Reply:
@jhedberg Good to know and it makes perfectly sense. I have changed to IS_ENABLED instead.